File: aoc.common.2022/Reads.fs
[<AutoOpen>]
module aoc.common.Reads

open System.Text.RegularExpressions

let readTokens (line: string) splitpattern =
    Regex
        .Replace(line, splitpattern, " ")
        .Trim()
        .Split(' ')

let readNumbers (line: string) =
    readTokens line @"[^\-\d]+" |> Array.map int64

let readNonNumbers (line: string) = readTokens line @"[\d]+"

let readNumber (line: string) =
    ((readTokens line @"[^\-\d]+") |> Array.map int64)[0]

let readMatrixOfNumbers (lines: string []) = lines |> Array.map readNumbers

File: aoc.common/Reads.cs
﻿using System;
using System.Text.RegularExpressions;
using System.Linq;


namespace aoc.common
{
    public static class Reads
    {
        public static string[] ReadTokens(string line, string splitPattern)
        {
            return Regex.Replace(line, splitPattern, " ").Trim().Split(' ');
        }

        public static long[] ReadNumbers(string line)
        {
            return ReadTokens(line, @"[^\-\d]+").Select(long.Parse).ToArray();
        }

        public static string[] ReadNonNumbers(string line)
        {
            return ReadTokens(line, @"[\d]+");
        }

        public static long ReadNumber(string line)
        {
            return ReadTokens(line, @"[^\-\d]+").Select(long.Parse).First();
        }

        public static long[][] ReadMatrixOfNumbers(string[] lines)
        {
            return lines.Select(ReadNumbers).ToArray();
        }
    }

}
